clang/lib/Format/UnwrappedLineParser.cpp

```diff
@@ -3117,9 +3117,16 @@ void UnwrappedLineParser::parseForOrWhileLoop(bool HasParens) {
       FormatTok->setFinalizedType(TT_ConditionLParen);
     parseParens();
   }
-  // Event control.
-  if (Style.isVerilog())
+
+  if (Style.isVerilog()) {
+    // Event control.
     parseVerilogSensitivityList();
+  } else if (Style.AllowShortLoopsOnASingleLine && FormatTok->is(tok::semi) &&
```

Comment:
Is `while (true) /*comment*/;` also short?

Reply:
Does it matter in the real world? Do we really need to decide whether the following is “short”?
```
/**/ while /**/ ( /**/ true /**/ ) /**/ ; //
```

Reply:
Just asking.
But I believe if you use `getPreviousNonComment()` it would be formatted as that.

Reply:
I didn't use `getPreviousNonComment()` because there are 3 more cases where we either can't or probably shouldn't merge the semicolon:

1.
```
while (1) //
  ;
```
2.
```
while (1) /**/
  ;
```
3.
```
while (1)
  /**/ ;
```
Like `while (1) /**/ ;`, case 3 probably doesn't matter in practice.

Reply:
Added test cases with comments before the null statement.

Reply:
BTW, neither `FormatTok->Previous` nor `FormatTok->getPreviousNonComment()` would have worked. 😉